src/proxy-bio.c

```diff
@@ -265,6 +276,51 @@ int http_connect (BIO *b)
     return -1;
   if ( (size_t) r != strlen(buf))
     return 0;
+
+  /* write proxy auth header, if applicable */
+  if (ctx->auth) {
+    c = ctx->auth;
+    r = strlen (PROXY_AUTH_HEADER_START);
+
+    /* make sure this header will fit in the buffer. */
+    /* base64-encoded size is 4/3 original size. */
+    int div_tmp = strlen(c) * 4; /* hacky way to get ceil(size*4/3) without -lmath. */
+    if ( !(sizeof (buf) >= r + (div_tmp/3 + (div_tmp % 3 != 0)) + 2)) {
+      /* buffer too small to accomadate header */
+      return -1;
+    }
+
+    strcpy (buf, PROXY_AUTH_HEADER_START);
+
+    /* base64-encode and write auth string */
+    while (*c) {
```

Comment:
why by hand?

Reply:
do you mean as opposed to a library? trying to avoid more recipes, etc.